table/fib-strategy-tree.go

```diff
@@ -80,8 +80,9 @@ func (f *fibStrategyTreeEntry) findLongestPrefixEntryEnc(name enc.Name) *fibStra
 func (f *FibStrategyTree) fillTreeToPrefixEnc(name enc.Name) *fibStrategyTreeEntry {
 	curNode := f.root.findLongestPrefixEntryEnc(name)
 	for depth := curNode.depth + 1; depth <= len(name); depth++ {
+		component, _ := enc.ComponentFromBytes(At(name, depth-1).Bytes())
```

Comment:
What is the purpose of this line?

Reply:
It's  a deep copy. I'm not really sure what is going on here: the underlying array gets deallocated / overwritten somehow if this isn't copied and it becomes garbage. If I remove the copies in this file:

```
root@0037b98ec2ac:/work/YaNFD# nfdc strategy set /svs/abcdefghijklmnopqrstuvwxyz /localhost/nfd/strategy/multicast
strategy-set prefix=/svs/abcdefghijklmnopqrstuvwxyz strategy=/localhost/nfd/strategy/multicast/v=1
root@0037b98ec2ac:/work/YaNFD# nfdc strategy set /svs/abcdef /localhost/nfd/strategy/multicast
strategy-set prefix=/svs/abcdef strategy=/localhost/nfd/strategy/multicast/v=1
root@0037b98ec2ac:/work/YaNFD# nfdc strategy list
prefix=/ strategy=/localhost/nfd/strategy/best-route/v=1
prefix=/%85h%0E/abcdef strategy=/localhost/nfd/strategy/multicast/v=1
prefix=/%85h%0E/abcdefk%27%07%25%08%09localhost%08%03nfd strategy=/tegy%08%09mul/cas/%20J%07%9F%D4t%E9%03/I%9C%F4Yr%99%A2oT/v=1
root@0037b98ec2ac:/work/YaNFD# 
```

Reply:
Thank you for clarifying. This seems to be a big issue. Maybe we should add a comment explaining this.